### PR TITLE
angular - Separate interfaces for each lifecycle hook

### PR DESCRIPTION
**Pull request by iocalebs**
Having one required hook per interface leads to greater type safety than
having a single IController interface with all properties optional.
Namely, the IController interface will not help you if you misspell a hook
name. (A slight improvement, admittedly, but an improvement nonetheless.)

This is closer to how the typings are done for lifecycle hooks in ng2: https://angular.io/docs/ts/latest/guide/lifecycle-hooks.html

### CODE REVIEW

**Comment by calebstdenis** [comment]
looks like I missed something enforced by tslint. Will fix

File: types/angular/index.d.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // Type definitions for Angular JS 1.6
 // Project: http://angularjs.org
-// Definitions by: Diego Vilar <http://github.com/diegovilar>, Georgii Dolzhykov <http://github.com/thorn0>
+// Definitions by: Diego Vilar <http://github.com/diegovilar>, Georgii Dolzhykov <http://github.com/thorn0>, Caleb St-Denis <https://github.com/calebstdenis>
```
Comment (thorn0) [inline types/angular/index.d.ts]: This means you want to become a reviewer for future PRs. Did you mean it?

Comment (calebstdenis) [inline types/angular/index.d.ts]: I have no problem with that. :) I know TypeScript well enough.

**Comment by thorn0** [comment]
Okay, would be nice to have one more reviewer.
Overall the PR looks good to me.

**Pull request merged**